src/main/java/org/truffleruby/core/hash/DeleteLastNode.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 2.0, or
+ * GNU General Public License version 2, or
+ * GNU Lesser General Public License version 2.1.
+ */
+package org.truffleruby.core.hash;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import org.truffleruby.language.RubyContextNode;
+
+import com.oracle.truffle.api.dsl.ImportStatic;
+import com.oracle.truffle.api.dsl.Specialization;
+
+@ImportStatic(HashGuards.class)
+public abstract class DeleteLastNode extends RubyContextNode {
+    /* For BucketHash, behaves almost identical to DeleteNode. Otherwise, deletes the most recently added key, because
+     * we are using the hash like a stack and guarantee that the last we added is the one we'll want to delete. When
+     * assertions are enabled checks that the deleted key is the one expected by the user. Does not handle blocks. A
+     * helper for detect_recursion. */
+
+    @Child private LookupEntryNode lookupEntryNode = new LookupEntryNode();
```

Comment:
Could you move this to the bucket specilization, as a `@Cached` argument?

Reply:
It seems to not compile when I do that?

Reply:
Ah, because there is no `create()` on `LookupEntryNode`.
You can use `@Cached("new()") LookupEntryNode lookupEntryNode` then.